src/libs/actions/Report.js

```diff
@@ -425,6 +417,65 @@ function removeOptimisticActions(reportID) {
     });
 }
 
+/**
+ * If an iouReport is open (has an IOU, but is not yet paid) then we maintain a link between it and the associated
+ * chatReport in Onyx, simplifying IOU data retrieval and reducing necessary API calls when displaying IOU components:
+ * - chatReport: {id: 123, iouReportID: 987, ...}
+ * - iouReport: {id: 987, chatReportID: 123, ...}
+ *
+ * This function allows us to fetch an iouReport without updating the linked chatReport, preventing the chatReport's
+ * 'iouReportID' value from being updated. As an example, this is desired when fetching historical reports, to avoid
+ * overwritting an open iouReportID with a closed iouReportID. If this was to occur, unpaid IOUs would not be
+ * highlighted to the user.
+ *
+ * If updating the chatReport's data is desired, use `fetchIOUReportByIDAndUpdateChatReportLink` instead.
+ *
+ * @param {Number} iouReportID - ID of the report we are fetching
+ * @param {Number} chatReportID - associated chatReportID, set as an iouReport field, but not used to maintain the link
+ */
+function fetchIOUReportByID(iouReportID, chatReportID) {
+    fetchIOUReport(iouReportID, chatReportID)
+        .then(iouReportObject => setLocalIOUReportData(iouReportObject));
+}
+
+/**
+ * If an iouReport is open (has an IOU, but is not yet paid) then we maintain a link between it and the associated
+ * chatReport in Onyx, simplifying IOU data retrieval and reducing necessary API calls when displaying IOU components:
+ * - chatReport: {id: 123, iouReportID: 987, ...}
+ * - iouReport: {id: 987, chatReportID: 123, ...}
+ *
+ * This link must remain in sync when the iouReport is modified. This function forces a link update after fetching the
```

Comment:
NAB, maybe I am alone here but referencing a "link" or "link update" instead of "sync reportIDs" makes this slightly difficult to follow.

Reply:
Replaced 'link'  with sync.

Reply:
I started to do this, but switching all references of 'link' to 'sync' loses some of the meaning. Instead, I'll selectively use sync in a few places where it makes the most sense.

Reply:
Maybe this is just arguing about semantics, but I don't think the link concept does us any favors. There is no "link". We are really just talking about whether or not to replace or remove a `reportID`. So why can't we just say that?

Reply:
I've now replaced any usage of link with sync, and mentioned specifically that it is the `iouReportID` value that must be kept in sync.